x-pack/test/api_integration/apis/management/rollup/index_patterns_extensions.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from 'expect.js';
+import querystring from 'querystring';
+
+import { registerHelpers } from './rollup.test_helpers';
+import { INDEX_TO_ROLLUP_MAPPINGS, INDEX_PATTERNS_EXTENSION_BASE_PATH } from './constants';
+import { getRandomString } from './lib';
+
+export default function ({ getService }) {
+  const supertest = getService('supertest');
+  const es = getService('es');
+
+  const {
+    createIndexWithMappings,
+    getJobPayload,
+    createJob,
+    cleanUp,
+  } = registerHelpers({ supertest, es });
+
+  describe('index patterns extension', () => {
+    describe('Fields for wildcards', () => {
+      const BASE_URI = `${INDEX_PATTERNS_EXTENSION_BASE_PATH}/_fields_for_wildcard`;
+
+      describe('query params validation', () => {
+        let uri;
+        let body;
+        let params;
+
+        it('"pattern" is required', async () => {
+          uri = `${BASE_URI}`;
+          ({ body } = await supertest.get(uri).expect(400));
+          expect(body.message).to.contain('"pattern" is required');
+        });
+
+        it('"params" is required', async () => {
+          params = { pattern: 'foo' };
+          uri = `${BASE_URI}?${querystring.stringify(params)}`;
+          ({ body } = await supertest.get(uri).expect(400));
+          expect(body.message).to.contain('"params" is required');
+        });
+
+        it('"params" must be an object', async () => {
+          params = { pattern: 'foo', params: 'bar' };
+          uri = `${BASE_URI}?${querystring.stringify(params)}`;
```

Comment:
How about adding a helper function for building the URI? I think this would make the code a bit easier to read and maintain.

```js
function buildUri(params = {}) {
  return `${BASE_URI}?${querystring.stringify(params)}`;
}

uri = buildUri({ pattern: 'foo', params: 'bar' });
```